### PR TITLE
Utilize hosted IPFS node and fallback to Infura

### PR DESCRIPTION
**Pull request by mbeacom**
##### Description

The goal of this PR is to utilize our hosted IPFS nodes over Infura and fallback to Infura if connectivity fails while introducing some logging.

##### TODO

Before merging this, we need to:

- [x] Spin up IPFS nodes on k8s for staging and prod
- [x] Test on staging

##### Checklist

- [x] linter status: 100% pass
- [x] changes don't break existing behavior
- [x] commit message follows [commit guidelines](https://github.com/gitcoinco/web/blob/master/docs/CONTRIBUTING.md#step-4-commit)

##### Affected core subsystem(s)
ipfs, bounties

### CODE REVIEW

**Comment by owocki** [comment]
i heard from @mbeylin that infura auto-pins everything through their infrastructure..

i wonder if we'll need to do the same or explicityly call `pin()` on each thing we uplaod

File: docker-compose.yml [hunk 1]
```diff
@@ -9,17 +9,17 @@ services:
     volumes:
       - pgdata:/var/lib/postgresql/data
 
-  pgweb:
-    image: sosedoff/pgweb
-    environment:
-        - DATABASE_URL=postgres://postgres:postgres@db:5432?sslmode=disable
```
Comment (owocki) [inline docker-compose.yml]: did you mean to remove the pgweb image?

File: app/app/settings.py [hunk 2]
```diff
@@ -408,6 +408,15 @@
 COLO_ACCOUNT_ADDRESS = env('COLO_ACCOUNT_ADDRESS', default='')  # TODO
 COLO_ACCOUNT_PRIVATE_KEY = env('COLO_ACCOUNT_PRIVATE_KEY', default='')  # TODO
 
+IPFS_HOST = env('IPFS_HOST', default='ipfs')
```
Comment (owocki) [inline app/app/settings.py]: shouldnt this default to infura?

File: docker-compose.yml (hunk 1 above)
Comment (mbeacom) [reply docker-compose.yml]: Yes.  Unless there is a lot of demand for this read-only web ui, we might as well strike it from the `docker-compose.yml`.

**Comment by mbeacom** [comment]
@owocki Think I should write a migration to pin all `IPFS` hashes in our system?

File: app/app/settings.py (hunk 2 above)
Comment (mbeacom) [reply app/app/settings.py]: We can definitely default to infura, if you want.  But I figured we could default to the local docker container of IPFS (all ipfs data will persist just as long as the DB data, so this should suffice and reduce latency) and use our hosted IPFS nodes as the primary on staging/live.

**Comment by owocki** [comment]
> @owocki Think I should write a migration to pin all IPFS hashes in our system?

it can't hurt, though i wonder if it's worth the effort given that infura is supposedly pinning them.

File: docker-compose.yml (hunk 1 above)
Comment (SaptakS) [reply docker-compose.yml]: I have always wondered if anyone uses this, because I usually log into the pgsql image and use psql to check db related stuff.

Comment (thelostone-mc) [reply docker-compose.yml]: I actually used it! :P

**Pull request merged**